proposals/collection-expressions-better-conversion.md

```diff
@@ -23,116 +23,85 @@ We add a new definition for ***better collection conversion from expression***,
+| `ReadOnlySpan<InterpolatedStringHandler>` | `ReadOnlySpan<string>` | `[$"{1}"]` | `[Interpolated String Handler]` | `[Identity]` | `CE₁ᵢ` is better | `ReadOnlySpan<InterpolatedStringHandler>` is picked |
+| `ReadOnlySpan<InterpolatedStringHandler>` | `ReadOnlySpan<string>` | `[$"{"blah"}"]` | `[Interpolated String Handler]` | `[Identity]` - But constant | `CE₂ᵢ` is better | `ReadOnlySpan<string>` is picked |
+| `ReadOnlySpan<string>` | `ReadOnlySpan<FormattableString>` | `[$"{1}"]` | `[Identity]` | `[Interpolated String]` | `CE₂ᵢ` is better | `ReadOnlySpan<string>` is picked |
+| `ReadOnlySpan<string>` | `ReadOnlySpan<FormattableString>` | `[$"{1}", (FormattableString)null]` | Not applicable | `[Interpolated String, Identity]` | `T₁` isn't applicable | `ReadOnlySpan<FormattableString>` is picked |
+| `HashSet<short>` | `Span<long>` | `[1, 2]` | `[Implicit Constant, Implicit Constant]` | `[Implicit Numeric, Implicit Numeric]` | `CE₁ᵢ` is better | `HashSet<short>` is picked |
+| `HashSet<long>` | `Span<short>` | `[1, 2]` | `[Implicit Numeric, Implicit Numeric]` | `[Implicit Constant, Implicit Constant]` | `CE₂ᵢ` is better | `Span<short>` is picked |
 
 ## Open questions
 
-### Should we include user-defined conversions in the compatibility check between `T'₁` and `T'₂`?
+### How far should we prioritize `ReadOnlySpan`/`Span` over other types?
 
-For example, should this work?
+As specified today, the following overloads would be ambiguous:
 
 ```cs
-C<int>.M([1, 2]);
+C.M1(["Hello world"]); // Ambiguous, no tiebreak between ROS and List
+C.M2(["Hello world"]); // Ambiguous, no tiebreak between Span and List
 
-class C<T> : IEnumerable<T>
-{
-    public static implicit operator Span<T>(C<T> t) => new T[];
-
-    public static void M(C<int> c) {}
-    public static void M(Span<int> c) {}
+C.M3(["Hello world"]); // Ambiguous, no tiebreak between ROS and MyList.
 
-    // Implementation of IEnumerable<T> and Add(T)
-}
-```
+C.M4(["Hello", "Hello"]); // Ambiguous, no tiebreak between ROS and HashSet. Created collections have different contents
 
-The element types are the same, so it's a simply a question of whether we recognize that there's user-defined relationship between `C<T>` and `Span<T>`; if we include the conversion, then `C<T>` is more specific than `Span<T>`, and
-that's the one we choose. Otherwise, we say this is ambiguous.
+class C
+{
+    public static void M1(ReadOnlySpan<string> ros) {}
+    public static void M1(List<string> list) {}
 
-### How far should we go in comparing base types
+    public static void M2(Span<string> ros) {}
+    public static void M2(List<string> list) {}
 
-Consider this scenario:
+    public static void M3(ReadOnlySpan<string> ros) {}
+    public static void M3(MyList<string> list) {}
 
-```cs
-class C<T> : IEnumerable<int> { public void Add(int i) { } }
+    public static void M4(ReadOnlySpan<string> ros) {}
+    public static void M4(HashSet<string> hashset) {}
```

Comment:
Consider removing the extra examples and using `M1` and `M2`. (Are any changes to the rules likely to affect `M3` or `M4` differently?)

Reply:
Yes, I think they are potentially going to affect them differently. I included all the examples I think the LDM needs to see to make a decision.